### PR TITLE
Fixed most deprecation warnings

### PR DESCRIPTION
**Pull request by achesak**

### CODE REVIEW

**Comment by achesak** [comment]
Scratch that I broke something...

**Comment by achesak** [comment]
I got a little overzealous with one of those changes. Everything working properly now.

**Comment by dom96** [comment]
Nice! I will postpone merging this until a new version of Nim is released.

**Comment by achesak** [comment]
Fixed more deprecation warnings as of the latest version of Nim.

File: processes.nim [hunk 1]
```diff
@@ -335,7 +335,7 @@ proc dispatchTasks(tasks: int, started: var bool, p: var Process, o: var Stream)
       if not started:
         let (bin, args) = cmdToArgs(task.command)
         p = startProcess(bin, task.workDir, args,
-                         options = {poStdErrToStdOut, poUseShell, poInteractive})
+                         options = {poStdErrToStdOut, poUsePath, poInteractive})
```
Comment (dom96) [inline processes.nim]: Please don't change this. Some people may wish to execute shell commands.

Comment (achesak) [inline processes.nim]: As per osproc module, "poUseShell" is a deprecated alias for "poUsePath".

Comment (dom96) [inline processes.nim]: I see

**Pull request merged**